dense_vector/track.py

```diff
@@ -197,6 +208,7 @@ async def __call__(self, es, params):
                         "num_candidates": num_candidates,
                     },
                     "_source": False,
+                    "profile": True,
```

Comment:
Have you tested this to see the latency impacts locally? I would be surprised if this didn't slow things down at all.

Reply:
> this slows down queries significantly. We shouldn't do this in a rally track :(.

This thing calculates recall, totally separate from the search stuff where we track query metrics. It's ok if the queries are a bit slow here; we're not keeping an eye on each one's timing. Just check out the extra time it takes on to the overall run, and that's all we need to worry about.

Reply:
I missed that this was the recall runner. Putting profile there is probably ok. ++

Reply:
I've been trying to complete a couple of runs since yesterday, but I'm facing some issues around the `force-merge` action. Will report the numbers as soon as I have them.

Reply:
Was finally able to run a couple of tests :) 

Differences are pretty small with overall runtime increasing from  ~52minutes to ~58minutes

**profile=true**
```
|                                                 Min Throughput |                            knn-recall-10-100 |      0.12        |  ops/s |
|                                                Mean Throughput |                            knn-recall-10-100 |      0.12        |  ops/s |
|                                              Median Throughput |                            knn-recall-10-100 |      0.12        |  ops/s |
|                                                 Max Throughput |                            knn-recall-10-100 |      0.12        |  ops/s |
|                                       100th percentile latency |                            knn-recall-10-100 |   8324.62        |     ms |
|                                  100th percentile service time |                            knn-recall-10-100 |   8324.62        |     ms |
|                                                     error rate |                            knn-recall-10-100 |      0           |      % |
|                                                 Min Throughput |                          knn-recall-100-1000 |      0.04        |  ops/s |
|                                                Mean Throughput |                          knn-recall-100-1000 |      0.04        |  ops/s |
|                                              Median Throughput |                          knn-recall-100-1000 |      0.04        |  ops/s |
|                                                 Max Throughput |                          knn-recall-100-1000 |      0.04        |  ops/s |
|                                       100th percentile latency |                          knn-recall-100-1000 |  22894.2         |     ms |
|                                  100th percentile service time |                          knn-recall-100-1000 |  22894.2         |     ms |
|                                                     error rate |                          knn-recall-100-1000 |      0           |      % |
|                                                 Min Throughput |                             knn-recall-10-10 |      0.12        |  ops/s |
|                                                Mean Throughput |                             knn-recall-10-10 |      0.12        |  ops/s |
|                                              Median Throughput |                             knn-recall-10-10 |      0.12        |  ops/s |
|                                                 Max Throughput |                             knn-recall-10-10 |      0.12        |  ops/s |
|                                       100th percentile latency |                             knn-recall-10-10 |   8100.69        |     ms |
|                                  100th percentile service time |                             knn-recall-10-10 |   8100.69        |     ms |
|                                                     error rate |                             knn-recall-10-10 |      0           |      % |
|                                                 Min Throughput |                           knn-recall-100-100 |      0.08        |  ops/s |
|                                                Mean Throughput |                           knn-recall-100-100 |      0.08        |  ops/s |
|                                              Median Throughput |                           knn-recall-100-100 |      0.08        |  ops/s |
|                                                 Max Throughput |                           knn-recall-100-100 |      0.08        |  ops/s |
|                                       100th percentile latency |                           knn-recall-100-100 |  11763           |     ms |
|                                  100th percentile service time |                           knn-recall-100-100 |  11763           |     ms |
|                                                     error rate |                           knn-recall-100-100 |      0           |      % |
|                                                 Min Throughput |                         knn-recall-1000-1000 |      0.03        |  ops/s |
|                                                Mean Throughput |                         knn-recall-1000-1000 |      0.03        |  ops/s |
|                                              Median Throughput |                         knn-recall-1000-1000 |      0.03        |  ops/s |
|                                                 Max Throughput |                         knn-recall-1000-1000 |      0.03        |  ops/s |
|                                       100th percentile latency |                         knn-recall-1000-1000 |  37751.3         |     ms |
|                                  100th percentile service time |                         knn-recall-1000-1000 |  37751.3         |     ms |
|                                                     error rate |                         knn-recall-1000-1000 |      0           |      % |
```

**profile=false**
```
|                                             Min Throughput |                            knn-recall-10-100 |      0.14        |  ops/s |
|                                                Mean Throughput |                            knn-recall-10-100 |      0.14        |  ops/s |
|                                              Median Throughput |                            knn-recall-10-100 |      0.14        |  ops/s |
|                                                 Max Throughput |                            knn-recall-10-100 |      0.14        |  ops/s |
|                                       100th percentile latency |                            knn-recall-10-100 |   6933.46        |     ms |
|                                  100th percentile service time |                            knn-recall-10-100 |   6933.46        |     ms |
|                                                     error rate |                            knn-recall-10-100 |      0           |      % |
|                                                 Min Throughput |                          knn-recall-100-1000 |      0.05        |  ops/s |
|                                                Mean Throughput |                          knn-recall-100-1000 |      0.05        |  ops/s |
|                                              Median Throughput |                          knn-recall-100-1000 |      0.05        |  ops/s |
|                                                 Max Throughput |                          knn-recall-100-1000 |      0.05        |  ops/s |
|                                       100th percentile latency |                          knn-recall-100-1000 |  20553.5         |     ms |
|                                  100th percentile service time |                          knn-recall-100-1000 |  20553.5         |     ms |
|                                                     error rate |                          knn-recall-100-1000 |      0           |      % |
|                                                 Min Throughput |                             knn-recall-10-10 |      0.19        |  ops/s |
|                                                Mean Throughput |                             knn-recall-10-10 |      0.19        |  ops/s |
|                                              Median Throughput |                             knn-recall-10-10 |      0.19        |  ops/s |
|                                                 Max Throughput |                             knn-recall-10-10 |      0.19        |  ops/s |
|                                       100th percentile latency |                             knn-recall-10-10 |   5295.42        |     ms |
|                                  100th percentile service time |                             knn-recall-10-10 |   5295.42        |     ms |
|                                                     error rate |                             knn-recall-10-10 |      0           |      % |
|                                                 Min Throughput |                           knn-recall-100-100 |      0.12        |  ops/s |
|                                                Mean Throughput |                           knn-recall-100-100 |      0.12        |  ops/s |
|                                              Median Throughput |                           knn-recall-100-100 |      0.12        |  ops/s |
|                                                 Max Throughput |                           knn-recall-100-100 |      0.12        |  ops/s |
|                                       100th percentile latency |                           knn-recall-100-100 |   8369.92        |     ms |
|                                  100th percentile service time |                           knn-recall-100-100 |   8369.92        |     ms |
|                                                     error rate |                           knn-recall-100-100 |      0           |      % |
|                                                 Min Throughput |                         knn-recall-1000-1000 |      0.03        |  ops/s |
|                                                Mean Throughput |                         knn-recall-1000-1000 |      0.03        |  ops/s |
|                                              Median Throughput |                         knn-recall-1000-1000 |      0.03        |  ops/s |
|                                                 Max Throughput |                         knn-recall-1000-1000 |      0.03        |  ops/s |
|                                       100th percentile latency |                         knn-recall-1000-1000 |  33339           |     ms |
|                                  100th percentile service time |                         knn-recall-1000-1000 |  33339           |     ms |
|                                                     error rate |                         knn-recall-1000-1000 |      0           |      % |
```

Reply:
++ Good stuff, thank you for the due diligence!